app/models/daos/UserDAO.scala

```diff
@@ -0,0 +1,86 @@
+package models.daos
+
+import scala.concurrent.Future
+
+import anorm._
+import anorm.SqlParser.get
+import com.mohiva.play.silhouette.api.LoginInfo
+import javax.inject.Inject
+import play.api.db.DBApi
+
+import models.{ DatabaseExecutionContext, User }
+import models.daos.UserDAO._
+
+class UserDAO @Inject()(dbapi: DBApi, ec: DatabaseExecutionContext) {
+  private val db = dbapi.database("default")
+
+  def find(loginInfo: LoginInfo) = Future {
+    val user: Option[User] = db.withConnection { implicit c =>
+      val email = loginInfo.providerKey
+      val result = SQL(
+        s"""
+          | SELECT id, first, last, email, type AS userType
+          | FROM users
+          | WHERE email = {email}
+        """.stripMargin).on("email" -> email).as(userRowParser.singleOpt)
+      result
+    }
+    user
+  }(ec)
+
+  def find(userID: Int) = Future {
+    db.withConnection { implicit c =>
+      SQL(
+        """
+          | SELECT id, first, last, email, type AS userType
+          | FROM user
+          | WHERE id = {userId}
+        """.stripMargin)
+        .on("userId" -> userID)
+        .as(userRowParser.single)
+    }
+  }(ec)
+
+  def save(user: User): Future[User] = Future {
+    val userId = db.withConnection { implicit conn =>
+      user match {
+        case User(_, Some(first), credentialId, Some(last), userType, Some(email)) =>
+          val query = s"""
```

Comment:
Should we try to be more consistent about defining intermediate values in these methods? Currently, `query` is variably: a `String`, a `SimpleSql`, not defined, called `result`.